### ISSUE TITLE
3.0.8 => 3.0.9 TypeError: Cannot read property 'fileName' of undefined

### ISSUE THREAD

**Issue by baywet** [monounity/karma-typescript#226]
Hi,
Our builds broke overnight because of the new version.  
So far we locked down to 3.0.8 to workaround the issue.  
Error message
```
19 12 2017 12:36:06.603:ERROR [karma]: TypeError: Cannot read property 'fileName' of undefined
    at C:\sources\gitttl\MeetingWorkspace\node_modules\karma-typescript\dist\compiler\compiler.js:105:46
    at Array.forEach (native)
    at Compiler.outputDiagnostics (C:\sources\gitttl\MeetingWorkspace\node_modules\karma-typescript\dist\compiler\compiler.js:104:21)
    at Compiler.runDiagnostics (C:\sources\gitttl\MeetingWorkspace\node_modules\karma-typescript\dist\compiler\compiler.js:97:14)
    at Compiler.compileProject (C:\sources\gitttl\MeetingWorkspace\node_modules\karma-typescript\dist\compiler\compiler.js:49:14)
    at Compiler.<anonymous> (C:\sources\gitttl\MeetingWorkspace\node_modules\karma-typescript\dist\compiler\compiler.js:28:23)
    at invokeFunc (C:\sources\gitttl\MeetingWorkspace\node_modules\lodash\lodash.js:10350:23)
    at trailingEdge (C:\sources\gitttl\MeetingWorkspace\node_modules\lodash\lodash.js:10397:18)
    at Timeout.timerExpired [as _onTimeout] (C:\sources\gitttl\MeetingWorkspace\node_modules\lodash\lodash.js:10385:18)
    at ontimeout (timers.js:386:14)
    at tryOnTimeout (timers.js:250:5)
    at Timer.listOnTimeout (timers.js:214:5)
```
karma.conf.js
```JS
// Karma configuration
// Generated on Tue Jun 13 2017 09:32:26 GMT-0400 (Eastern Daylight Time)
// https://www.npmjs.com/package/karma-tfs-reporter
module.exports = function(config) {
  config.set({

    // base path that will be used to resolve all patterns (eg. files, exclude)
    basePath: '',


    // frameworks to use
    // available frameworks: https://npmjs.org/browse/keyword/karma-adapter
    frameworks: ['mocha', 'chai', 'sinon', 'karma-typescript'],


    // list of files / patterns to load in the browser
    files: [
      'node_modules/core-js/client/core.min.js',
      'node_modules/jquery/dist/jquery.min.js', // add local jquery
      'node_modules/pickadate/lib/picker.js',
      'node_modules/office-ui-fabric-js/dist/lib/PickaDate.js',
      'node_modules/office-ui-fabric-js/dist/lib/scrollTo.js',
      'node_modules/office-ui-fabric-js/dist/js/fabric.min.js',
      'node_modules/loglevel/dist/loglevel.js',
      {pattern: 'src/**/*.ts'},
    ],


    // list of files to exclude
    exclude: [
      'node_modules/*'
    ],


    // preprocess matching files before serving them to the browser
    // available preprocessors: https://npmjs.org/browse/keyword/karma-preprocessor
    preprocessors: {
      './src/**/*.ts': ['karma-typescript'],
    },


    // test results reporter to use
    // possible values: 'dots', 'progress'
    // available reporters: https://npmjs.org/browse/keyword/karma-reporter
    reporters: ['progress', 'mocha', 'karma-typescript', 'junit'],
    karmaTypescriptConfig: {
      bundlerOptions: {
        resolve: {
          alias: {
            "exports-loader?fabric!office-ui-fabric-js/dist/js/fabric.min": "node_modules/office-ui-fabric-js/dist/js/fabric.min.js",
            "./picker.js": "node_modules/pickadate/lib/picker.js"
          }
        }
      },
      compilerOptions: {
        allowJs: true,
      },
      reports:
      {
          "html":  { directory: 'test/coverage', subdirectory: 'cobertura', },
          "cobertura": { directory: 'test/coverage', subdirectory: 'cobertura', filename: 'cobertura.xml' },
          "lcovonly": { directory: 'test/coverage', subdirectory: 'lcov', filename: 'lcov.info' },
      },
      tsconfig: './tsconfig.json'
    },
    junitReporter: {
      outputDir: 'test/', // results will be saved as $outputDir/$browserName.xml
      outputFile: 'test-results.xml', // if included, results will be saved as $outputDir/$browserName/$outputFile
      suite: 'karma', // suite will become the package name attribute in xml testsuite element
      useBrowserName: true, // add browser name to report and classes names
    },

    // web server port
    port: 9876,


    // enable / disable colors in the output (reporters and logs)
    colors: true,


    // level of logging
    // possible values: config.LOG_DISABLE || config.LOG_ERROR || config.LOG_WARN || config.LOG_INFO || config.LOG_DEBUG
    logLevel: config.LOG_DEBUG,


    // enable / disable watching file and executing tests whenever any file changes
    autoWatch: true,


    // start these browsers
    // available browser launchers: https://npmjs.org/browse/keyword/karma-launcher
    // browsers: ['Chrome', 'PhantomJS', 'IE', 'Firefox', 'Safari'],
    browsers: ['PhantomJS'],


    // Continuous Integration mode
    // if true, Karma captures browsers, runs the tests and exits
    singleRun: false,

    // Concurrency level
    // how many browser should be started simultaneous
    concurrency: Infinity
  })
}
```
tsconfig.json
```JSON
{
  "compilerOptions": {
    "target": "ES5",
    "lib": [
      "dom",
      "dom.iterable",
      "es2015",
      "scripthost"
    ],
    "sourceMap": true,
    "typeRoots": [ "node_modules/@types/" ],
    "declaration": true,
    "noFallthroughCasesInSwitch": true,
    "noImplicitAny": true,
    "noImplicitReturns": true,
    "noImplicitThis": true,
    "noUnusedLocals": true,
    "noUnusedParameters": true,
    "preserveConstEnums": true,
    "removeComments": false,
    "outDir": "dist",
    "skipDefaultLibCheck": true,
    "skipLibCheck": true
  },
  "compileOnSave": false,
  "exclude": [
    ".vscode",
    "node_modules",
    "dist",
    "webpack",
    "test",
    "webpack.config.js",
    "karma.conf.js",
    "postcss.config.js",
    "src/templates"
  ]
}
```
packages.json (partial)
```JSON
 "dependencies": {
    "@types/moment-timezone": "^0.2.34",
    "copy-html-to-clipboard": "^4.0.1",
    "datatables.net": "^1.10.15",
    "datatables.net-dt": "^1.10.15",
    "datatables.net-rowreorder": "^1.2.0",
    "datatables.net-rowreorder-dt": "^1.2.0",
    "datatables.net-select": "^1.2.2",
    "datatables.net-select-dt": "^1.2.2",
    "debug": "^3.1.0",
    "delay": "^2.0.0",
    "hellojs": "^1.14.1",
    "i18next": "^8.4.2",
    "i18next-browser-languagedetector": "^2.1.0",
    "i18next-ko": "^3.0.1",
    "jquery": "^3.2.1",
    "jquery-editable-select": "^2.2.5",
    "jstz": "^1.0.10",
    "knockout": "^3.4.2",
    "knockout-mapping": "^2.6.0",
    "knockout-postbox": "^0.5.0",
    "lodash": "^4.17.4",
    "loglevel": "^1.6.0",
    "mime": "^2.0.3",
    "moment": "^2.19.2",
    "office-ui-fabric-core": "^7.1.1",
    "office-ui-fabric-js": "^1.4.0",
    "parsejson": "0.0.3",
    "pickadate": "^3.5.6",
    "sp-pnp-js": "^3.0.1",
    "whatwg-fetch": "^2.0.3"
  },
  "devDependencies": {
    "@microsoft/microsoft-graph-types": "^1.0.1",
    "@types/chai": "^4.0.5",
    "@types/core-js": "^0.9.41",
    "@types/datatables.net": "^1.10.7",
    "@types/datatables.net-buttons": "^1.4.0",
    "@types/datatables.net-fixedheader": "^3.1.3",
    "@types/datatables.net-rowreorder": "^1.1.1",
    "@types/datatables.net-select": "^1.2.3",
    "@types/hellojs": "^1.15.0",
    "@types/i18next": "^8.4.0",
    "@types/i18next-browser-languagedetector": "0.0.31",
    "@types/jquery": "^3.2.16",
    "@types/jquery-editable-select": "^2.2.0",
    "@types/knockout": "^3.4.47",
    "@types/knockout.mapping": "^2.0.30",
    "@types/knockout.postbox": "0.0.28",
    "@types/lodash": "^4.14.85",
    "@types/loglevel": "^1.4.29",
    "@types/mocha": "^2.2.44",
    "@types/moment-timezone": "^0.2.34",
    "@types/node": "^7.0.48",
    "@types/pickadate": "^3.5.29",
    "@types/sharepoint": "^2016.0.2",
    "@types/sinon": "^2.3.7",
    "chai": "^4.1.2",
    "chai-as-promised": "^7.0.0",
    "core-js": "^2.4.1",
    "css-loader": "^0.28.1",
    "exports-loader": "^0.6.4",
    "extract-text-webpack-plugin": "^3.0.2",
    "file-loader": "^1.1.5",
    "fork-ts-checker-webpack-plugin": "^0.2.7",
    "html-loader": "^0.5.1",
    "jquery-editable-select": "^2.2.5",
    "karma": "^1.7.1",
    "karma-chai": "^0.1.0",
    "karma-coverage": "^1.1.1",
    "karma-junit-reporter": "^1.2.0",
    "karma-mocha": "^1.3.0",
    "karma-mocha-reporter": "^2.2.5",
    "karma-phantomjs-launcher": "^1.0.4",
    "karma-remap-coverage": "^0.1.4",
    "karma-sinon": "^1.0.5",
    "karma-sourcemap-loader": "^0.3.7",
    "karma-typescript": "3.0.8",
    "karma-typescript-preprocessor2": "^1.2.1",
    "karma-webpack": "^2.0.6",
    "mocha": "^4.0.1",
    "node-sass": "^4.7.1",
    "phantomjs-prebuilt": "^2.1.16",
    "pickadate": "^3.5.6",
    "postcss": "^6.0.14",
    "postcss-loader": "^2.0.8",
    "raw-loader": "^0.5.1",
    "resolve-url-loader": "^2.2.0",
    "sass-lint": "^1.12.1",
    "sass-loader": "^6.0.3",
    "sinon": "^4.1.2",
    "source-map-loader": "^0.2.1",
    "style-loader": "^0.19.0",
    "ts-loader": "^3.1.1",
    "tslint": "^5.1.0",
    "tslint-eslint-rules": "^4.0.0",
    "typescript": "^2.4.2",
    "uglifyjs-webpack-plugin": "^1.1.0",
    "url-loader": "^0.6.2",
    "webpack": "^3.8.1",
    "webpack-dev-server": "^2.9.4",
    "webpack-merge": "^4.1.0",
    "whatwg-fetch": "^2.0.3"
  },
```

**Comment by erikbarke** [comment]
It looks like the compiler barfed when trying to collect configuration errors, there's a fix here: #227 ☺️

**Comment by erikbarke** [comment]
Fixed in #227, closing.
